packages/cactus-plugin-ledger-connector-polkadot/src/test/typescript/integration/invoke-ink-contract.test.ts

```diff
@@ -0,0 +1,179 @@
+  beforeAll(async () => {
+    await ledger.start();
+    expect(ledger).toBeTruthy();
+    keychainEntryKey = uuidv4();
+    keychainEntryValue = "//Bob";
+    keychainPlugin = new PluginKeychainMemory({
+      instanceId: uuidv4(),
+      keychainId: uuidv4(),
+      // pre-provision keychain with mock backend holding the private key of the
+      // test account that we'll reference while sending requests with the
+      // signing credential pointing to this keychain entry.
+      backend: new Map([[keychainEntryKey, keychainEntryValue]]),
+      logLevel,
+    });
+    const connectorOptions: IPluginLedgerConnectorPolkadotOptions = {
+      logLevel: logLevel,
+      pluginRegistry: new PluginRegistry({ plugins: [keychainPlugin] }),
+      wsProviderUrl: DEFAULT_WSPROVIDER,
+      instanceId: instanceId,
+    };
+    plugin = new PluginLedgerConnectorPolkadot(connectorOptions);
+    await plugin.createAPI();
+
+    const listenOptions: IListenOptions = {
+      hostname: "0.0.0.0",
+      port: 0,
+      server,
+    };
+    addressInfo = await Servers.listen(listenOptions);
+    ({ address, port } = addressInfo);
+    apiHost = `http://${address}:${port}`;
+    apiConfig = new Configuration({ basePath: apiHost });
+    apiClient = new PolkadotApi(apiConfig);
+    await plugin.registerWebServices(expressApp);
+    await plugin.getOrCreateWebServices();
+  });
+  beforeAll(async () => {
+    const rawWasm = fs.readFileSync(
+      "packages/cactus-plugin-ledger-connector-polkadot/src/test/rust/fixtures/ink/flipper.wasm",
+    );
+    const result = await apiClient.deployContractInk({
+      wasm: rawWasm.toString("base64"),
+      metadata: JSON.stringify(metadata),
+      gasLimit: gasLimit,
+      storageDepositLimit: null,
+      salt: new Uint8Array().toString(),
+      web3SigningCredential: { type: "MNEMONIC_STRING", mnemonic: "//Alice" },
+      params: [false],
+    });
+    expect(result).toBeTruthy();
+    expect(result.data.success).toBeTrue;
+    expect(result.data.contractAddress).toBeTruthy();
+    if (!result.data.contractAddress) {
+      throw new Error("contract address cannot be undefined");
+    }
+    contractAddress = result.data.contractAddress
+  });
+  test("invalid methodName contract invocation", async () => {
+    const result = apiClient.invokeContract({
+      invocationType: "SEND",
+      contractAddress,
+      gasLimit,
+      metadata: JSON.stringify(metadata),
+      methodName: "invalid",
+      accountAddress: "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY", //Alice account address
+      web3SigningCredential: {
+        type: "CACTUS_KEYCHAIN_REF",
+        keychainEntryKey: keychainEntryKey,
+        keychainId: keychainPlugin.getKeychainId(),
+      },
+    });
+    await expect(result).rejects.toHaveProperty(["response", "status"], 400);
+  });
+  test("query ink! contract", async () => {
+    const result = await apiClient.invokeContract({
+      invocationType: "QUERY",
```

Comment:
@AnmolBansalDEV Please make sure to define constants like this in the openapi.json file which will then also make it available for import via the generated typescript code and then you can use those code constants instead of magic strings.

Reply:
updated all the constants to openapi.json's code constants